core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -898,7 +939,7 @@ public boolean isFollowing() {
                 }
                 // Find the parent key assuming this is not the root key, and not an account key for a watching chain.
                 DeterministicKey parent = null;
-                if (!path.isEmpty() && !isWatchingAccountKey) {
+                if (!path.isEmpty() && !isWatchingAccountKey & !isSpendingKey) {
```

Comment:
Binary `&`? Unless `!isSpendingKey` has to get evaluated no matter what, use `&&` here and even if `isSpendingKey` had side effects, I would find it confusing to rely on [this particularity](https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.22.2) of the binary `&` in a logical expression.

Reply:
Looks unintended to me.

Reply:
Yes, that is a typo (`&` instead of `&&`).